sysl2/sysl/syslutil/chroot_fs.go

```diff
@@ -15,58 +17,152 @@ type ChrootFs struct {
-	return fs.fs.Create(fs.join(name))
+	data, err := fs.wrapCallWithData(name,
+		func(fixedPath string) (interface{}, error) {
+			return fs.fs.Create(fixedPath)
+		})
+	if err != nil {
+		return nil, err
+	}
+	return data.(afero.File), err
 }
 
 func (fs *ChrootFs) Mkdir(name string, perm os.FileMode) error {
-	return fs.fs.Mkdir(fs.join(name), perm)
+	return fs.wrapCall(name, func(fixedPath string) error {
+		return fs.fs.Mkdir(fixedPath, perm)
+	})
 }
 
 func (fs *ChrootFs) MkdirAll(path string, perm os.FileMode) error {
-	return fs.fs.MkdirAll(fs.join(path), perm)
+	return fs.wrapCall(path, func(fixedPath string) error {
+		return fs.fs.MkdirAll(fixedPath, perm)
+	})
 }
 
 func (fs *ChrootFs) Open(name string) (afero.File, error) {
-	return fs.fs.Open(fs.join(name))
+	data, err := fs.wrapCallWithData(name,
+		func(fixedPath string) (interface{}, error) {
+			return fs.fs.Open(fixedPath)
+		})
+	if err != nil {
+		return nil, err
+	}
+	return data.(afero.File), err
 }
 
 func (fs *ChrootFs) OpenFile(name string, flag int, perm os.FileMode) (afero.File, error) {
-	return fs.fs.OpenFile(fs.join(name), flag, perm)
+	data, err := fs.wrapCallWithData(name,
+		func(fixedPath string) (interface{}, error) {
+			return fs.fs.OpenFile(fixedPath, flag, perm)
+		})
+	if err != nil {
+		return nil, err
+	}
+	return data.(afero.File), err
 }
 
 func (fs *ChrootFs) Remove(name string) error {
-	return fs.fs.Remove(fs.join(name))
+	return fs.wrapCall(name, func(fixedPath string) error {
+		return fs.fs.Remove(fixedPath)
+	})
 }
 
 func (fs *ChrootFs) RemoveAll(path string) error {
-	return fs.fs.RemoveAll(fs.join(path))
+	return fs.wrapCall(path, func(fixedPath string) error {
+		return fs.fs.RemoveAll(fixedPath)
+	})
 }
 
 func (fs *ChrootFs) Rename(oldname, newname string) error {
-	return fs.fs.Rename(fs.join(oldname), fs.join(newname))
+	return fs.wrapCall(oldname, func(fixedPath string) error {
+		newFile, err := fs.join(newname)
+		if err != nil {
+			return err
+		}
+		return fs.fs.Rename(fixedPath, newFile)
+	})
 }
 
 func (fs *ChrootFs) Stat(name string) (os.FileInfo, error) {
-	return fs.fs.Stat(fs.join(name))
+	data, err := fs.wrapCallWithData(name,
+		func(fixedPath string) (interface{}, error) {
+			return fs.fs.Stat(fixedPath)
+		})
+	if err != nil {
+		return nil, err
+	}
+	return data.(os.FileInfo), err
 }
 
 func (fs *ChrootFs) Name() string {
 	return "ChrootFS"
 }
 
 func (fs *ChrootFs) Chmod(name string, mode os.FileMode) error {
-	return fs.fs.Chmod(fs.join(name), mode)
+	return fs.wrapCall(name, func(fixedPath string) error {
+		return fs.fs.Chmod(fixedPath, mode)
+	})
 }
 
 func (fs *ChrootFs) Chtimes(name string, atime time.Time, mtime time.Time) error {
-	return fs.fs.Chtimes(fs.join(name), atime, mtime)
+	return fs.wrapCall(name, func(fixedPath string) error {
+		return fs.fs.Chtimes(fixedPath, atime, mtime)
+	})
+}
+
+func (fs *ChrootFs) openAllowed(fullPath string) error {
+	absoluteRoot, err := filepath.Abs(fs.root)
```

Comment:
don't think so, it needs to check for the module in the CLI argument and any other import paths that the module imports

Reply:
I think the premise is that `filepath.Abs(fs.root)` will always return the same value. Whether that's strictly true or not, in practice its a good working assumption (arguably better than being technically correct). This is a good idea. In fact, `NewChrootFs` should call `Abs` and assign the result to `fs.root`.

Reply:
That was actually part of a previous iteration, but I removed it because it changed a lot of files (a lot of `NewChrootFs` calls are a function argument). Will refactor this.

Reply:
error handling added